File: Support Material/Labfiles/Solution/M16/CS/CustomerManagement/App_Code/CustomersWcfDS.cs
using System;
using System.Data.Services;
using System.Data.Services.Common;
using System.Collections.Generic;
using System.Linq;
using System.ServiceModel.Web;

public class CustomersWcfDS : DataService<CustomerManagementModel.Entities>
{
    // This method is called only once to initialize service-wide policies.
    public static void InitializeService(DataServiceConfiguration config)
    {
        config.SetEntitySetAccessRule("Countries", EntitySetRights.AllRead);
        config.SetEntitySetAccessRule("Customers", EntitySetRights.AllRead);
        config.DataServiceBehavior.MaxProtocolVersion = DataServiceProtocolVersion.V2;
    }
}

File: Support Material/Labfiles/Starter/M14/VB/CustomerManagement/App_Code/CustomersWcfDS.vb
Imports System.Data.Services
Imports System.Data.Services.Common
Imports System.Linq
Imports System.ServiceModel.Web

Public Class CustomersWcfDS
    Inherits DataService(Of CustomerManagementModel.Entities)

    ' This method is called only once to initialize service-wide policies.
    Public Shared Sub InitializeService(ByVal config As DataServiceConfiguration)
        config.SetEntitySetAccessRule("Countries", EntitySetRights.AllRead)
        config.SetEntitySetAccessRule("Customers", EntitySetRights.AllRead)
        config.DataServiceBehavior.MaxProtocolVersion = DataServiceProtocolVersion.V2
    End Sub

End Class
